sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/TypedAggregateExpression.scala

```diff
@@ -76,7 +76,7 @@ object TypedAggregateExpression {
         None,
         bufferSerializer,
         bufferEncoder.resolveAndBind().deserializer,
-        outputEncoder.serializer,
+        outputEncoder.objSerializer,
```

Comment:
to confirm, this is a un-related change and just clean up the code?

Reply:
This is required. Without this, the output schema of aggregator using Option[Product] as output encoder is not correct.